src/binaryen-c.cpp

```diff
@@ -3812,6 +3812,37 @@ uint32_t BinaryenGetMemorySegmentByteOffset(BinaryenModuleRef module,
   Fatal() << "non-constant offsets aren't supported yet";
   return 0;
 }
+bool BinaryenHasMemory(BinaryenModuleRef module) {
+  return ((Module*)module)->memory.exists;
+}
+BinaryenIndex BinaryenMemoryGetInitial(BinaryenModuleRef module) {
+  return ((Module*)module)->memory.initial;
+}
+bool BinaryenMemoryHasMax(BinaryenModuleRef module) {
+  return ((Module*)module)->memory.hasMax();
+}
+BinaryenIndex BinaryenMemoryGetMax(BinaryenModuleRef module) {
+  return ((Module*)module)->memory.max;
+}
+const char* BinaryenMemoryImportGetModule(BinaryenModuleRef module) {
+  auto& memory = ((Module*)module)->memory;
+  if (memory.imported()) {
+    return memory.module.c_str();
+  } else {
+    return "";
+  }
+}
+const char* BinaryenMemoryImportGetBase(BinaryenModuleRef module) {
+  auto& memory = ((Module*)module)->memory;
+  if (memory.imported()) {
+    return memory.base.c_str();
+  } else {
+    return "";
```

Comment:
I think it would make more sense to return `NULL` here and above? IIUC, empty string is a perfectly valid string for import modules and names.

OTOH, I see that there is plenty of precedent for using empty string in this context, so we probably shouldn't change that in this PR.

Reply:
Yeah I agree with you, but it looks like this is the convention used for other entities like tables, functions, etc. It's particularly annoying that from the JS perspective you get fields with empty strings for base/name when it would make way more sense for those fields to be undefined. That being said, unless you feel strongly I don't really want to change this expectation in this PR. Who knows what clients rely on the empty string behavior? Hyrum's law and all that.